quinn-h3/src/lib.rs

```diff
@@ -1,3 +1,74 @@
+//! HTTP/3 support for tokio.
+//!
+//! This library is an asynchronous implementation of the [`HTTP/3`] protocol, part of the [`quinn`]
+//! project: a [`QUIC`] transport protocol imlementation.
+//!
+//! This crate is built to integrate well into the rust `async` and [`tokio`] ecosystem. It is heavily
+//! inspired by the [`h2`] crate's API, seeking for easy integration aside of it in a variety of projects.
+//!
+//! # About HTTP/3
+//!
+//! `HTTP/3` is the upcoming third major version of the Hypertext Transfer Protocol. It's built on top
+//! of a new transport protocol over `UDP`: [`QUIC`]. It addresses the shortcomings that [`HTTP/2`]
+//! suffers from being built on top of `TCP`, such as [`Head-Of-Line blocking`], or handshake latencies.
+//! Thanks to `QUIC` streams multiplexing features over `UDP`, requests are made indempendent from each other
+//! at the data transimission level.
+//!
+//! Unlike HTTP/1, it exchanges headers in binary format, and compresses them with [`QPACK`].
+//!
+//! It also offers Server Push features similar to `HTTP/2`.
+//!
+//! The HTTP/3 specification has not yet been stabilized. The version currently implemented by this
+//! crate is [`draft-27`].
+//!
+//! # Crate overview
+//!
+//! This crate is split into two main [`client`] and [`server`] modules. Body handling is done
+//! through [`BodyReader`] and [`BodyWriter`] which respectively implement [`AsyncRead`] and
+//! [`AsyncWrite`].
+//!
+//! This crate makes an extensive use of hyperium's [`http`] crate, which helps constructing
+//! [`Request`]s and [`Response`]s compatible with all the `HTTP` versions out there.
+//!
+//! # Crate status
+//!
+//! This crate is in an experimental state. For now, it has multiple improvement vectors:
+//!
+//! * Though having been tested as compatible with a marjority of other HTTP/3 implementations,
+//!   `quinn-h3` does not implement all interoperability tests for the moment.
+//! * The `CONNECT` method and associated proxy features are not implemented.
+//! * Server `PUSH` is not implemented.
+//!
+//! # Getting started
+//!
+//! Give the examples a try:
+//!
+//! ```sh
+//! cargo run --example h3_server &
+//! cargo run --example h3_client
+//! ```
+//!
+//! [`HTTP/3`]: https://en.wikipedia.org/wiki/HTTP/3
+//! [`quinn`]: https://docs.rs/quinn
+//! [`tokio`]: https://docs.rs/tokio
+//! [`h2`]: https://docs.rs/h2
+//! [`QUIC`]: https://en.wikipedia.org/wiki/QUIC
```

Comment:
```suggestion
//! [QUIC]: https://en.wikipedia.org/wiki/QUIC
```

Reply:
Is there a difference between those ?

Reply:
My point is that we usually use the code spans (the thing in backticks) for names in code. QUIC is an acronym and protocol but not something that appears in our code. The same goes for UDP and HTTP. (Maybe I misunderstood how this link syntax works, but I was under the impression the backticks here are used to format the text as a code span.)

Reply:
The syntax accepts both with and without backquotes. But all crates documentations use the ```[`link`]``` syntax, so I took that as convention.

Reply:
Yes, but again -- that is for code identifiers. QUIC and UDP are not identifiers.

Reply:
Right, I didn't notice the rendering difference. 

I got that fixed along with other non-code enclosed with ` 

:)

Reply:
I don't think I'm seeing that show up in the PR yet?

Reply:
Oops, didn't push it. It's now in the PR.